apps/ledger-live-desktop/src/newArch/features/ModularDrawer/Web3AppWebview/AssetAndAccountDrawer.ts

```diff
@@ -0,0 +1,79 @@
+import { Account, AccountLike } from "@ledgerhq/types-live";
+import { Observable } from "rxjs";
+import { WalletAPIAccount } from "@ledgerhq/live-common/wallet-api/types";
+import { CryptoOrTokenCurrency } from "@ledgerhq/types-cryptoassets";
+import { setDrawer } from "~/renderer/drawers/Provider";
+import SelectAssetFlow from "../components/SelectAssetFlow";
+import SelectAccountDrawer from "~/renderer/drawers/DataSelector/SelectAccountDrawer";
+
+type Result = {
+  account: AccountLike;
+  parentAccount?: Account;
+};
+
+type DrawerParams = {
+  type: "promise" | "callback";
+  assetIds?: string[];
+  currencies?: CryptoOrTokenCurrency[];
+  accounts$?: Observable<WalletAPIAccount[]>;
+  includeTokens?: boolean;
+  onSuccess?: (account: AccountLike, parentAccount?: Account) => void;
+  onCancel?: () => void;
+};
+
+export function openAssetAndAccountDrawer(params: DrawerParams): Promise<Result> | void {
```

Comment:
I think it's a good idea to open the drawer from a callback however I think this callback implements too much logic which would be easier to put in the drawer itself.

Here is what I have in mind:

- Create a new component in `ModularDrawer/components/SelectAccountFlow.tsx` with the code:
   ```tsx
   function SelectAccountDrawer ({  assetIds, ...props }: SelectAccountDrawerProps) {
     const currencies = listAndFilterCurrencies({ currencies: assetIds, includeTokens: true });
     return <SelectAccountAndCurrencyDrawer {...props} currencies={currencies} />
   }
   ```
- Then here the `setDrawer` only calls `SelectAccountDrawer` not `SelectAssetFlow`. These 2 flow will share a lot of there UI but `SelectAssetFlow` shouldn't be called when we want to select an account.